examples/erc721/lib.rs

```diff
@@ -130,12 +129,18 @@ mod erc721 {
         /// Creates a new ERC-721 token contract.
         #[ink(constructor)]
         pub fn new() -> Self {
-            Self {
-                token_owner: Default::default(),
-                token_approvals: Default::default(),
-                owned_tokens_count: Default::default(),
-                operator_approvals: Default::default(),
-            }
+            ink_lang::codegen::initialize_contract(|_| {
+                Self {
+                    // TODO: How should this be properly initialized?
+                    //
+                    // Turns out that `token_owner` and `token_approvals` have the same type
+                    // signature and they're both Default initialized to ROOT_KEY [0u8; 32]. This
+                    // means that they end up operating on the same storage keys :(
+                    // token_approvals: Default::default(),
+                    token_approvals: Mapping::new([1u8; 32].into()),
```

Comment:
@Robbepop looks like we need another piece of "entropy" when [assigning storage keys](https://github.com/paritytech/ink/blob/78b45330e453500a759a2b53b7ab6f16bfb91caa/crates/storage/src/lazy/mapping.rs#L115) in the `Mapping`. 

I haven't been able to figure out what we could use though, do you have any ideas?

Reply:
I noticed this "bug" when before swapping the initialization routine to actually use `ink_lang::codegen::initialize_contract`. By using it the `Mapping`'s get intialized with different `offset_key`s, so false alarm 😅